main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala

```diff
@@ -64,6 +64,10 @@ object ErasedAst {
 
     case class NewObject(name: String, clazz: java.lang.Class[_], tpe: MonoType, methods: List[JvmMethod], loc: SourceLocation) extends Expr
 
+    case class Instanceof(exp: Expr, clazz: java.lang.Class[_], loc: SourceLocation) extends Expr {
```

Comment:
We should make this into an `IntrinsicOperator1` instead of an expression.

(We are reducing the "surface" of expressions in preparation for fine-grain call by value which we need for AE).

Reply:
Done.

You might want to check the order of constructors.